component/authenticator/src/main/java/org/wso2/carbon/identity/application/authenticator/totp/TOTPAuthenticator.java

```diff
@@ -286,8 +286,16 @@ protected void initiateAuthenticationRequest(HttpServletRequest request,
                         claims = TOTPKeyGenerator.generateClaims(UserCoreUtil.addDomainToName(username,
                                 authenticatingUser.getUserStoreDomain()), false, context);
                     }
-                    context.setProperty(TOTPAuthenticatorConstants.SECRET_KEY_CLAIM_URL,
-                            claims.get(TOTPAuthenticatorConstants.SECRET_KEY_CLAIM_URL));
+                    Map<String, String> claimProperties = TOTPUtil.getClaimProperties(tenantDomain,
+                            TOTPAuthenticatorConstants.SECRET_KEY_CLAIM_URL);
+                    // Context will have the decrypted secret key all the time.
+                    if (claimProperties.containsKey(TOTPAuthenticatorConstants.ENABLE_ENCRYPTION)) {
```

Comment:
Can you re-evaluate this logic? If the enabled encryption key is there, what does it mean?

Reply:
If the enable encryption attribute is there, that means encryption will happen at the listener level and not in the service. That means `claims` will contain the secret key as plain text. If the enable encryption attribute is not there that indicates encryption will be taking place at the service level. hence we need to decrypt the claim value